### PR TITLE
Add gemmx-dr

### PR DESCRIPTION
**Pull request by xiaoling-yi**
We add the latest features and the latest arch in this PR.

### CODE REVIEW

File: target/snitch_cluster/Makefile [hunk 1]
```diff
@@ -148,6 +148,25 @@ $(eval $(call generate_snax_gen,snax_streamer_gemmX))
...
+    VLT_BENDER  += -t snax_streamer_gemmX
+    VCS_BENDER  += -t snax_streamer_gemmX
+
+$(eval $(call generate_snax_gen,snax_data_reshuffler))
+
+	VSIM_BENDER += -t snax_data_reshuffler
+	VLT_BENDER  += -t snax_data_reshuffler
+	VCS_BENDER  += -t snax_data_reshuffler
+
+endif
```
Comment (rgantonio) [inline target/snitch_cluster/Makefile:168]:
```suggestion
ifeq (${CFG_OVERRIDE}, cfg/snax-streamer-gemmX-data-reshuffler.hjson)
 
$(eval $(call generate_snax_gen,snax_streamer_gemmX))

    SNAX_GEMMX_ROOT ?= $(ROOT)/hw/chisel_acc
    include $(SNAX_GEMMX_ROOT)/Makefile
 
    VSIM_BENDER += -t snax_streamer_gemmX
    VLT_BENDER  += -t snax_streamer_gemmX
    VCS_BENDER  += -t snax_streamer_gemmX

$(eval $(call generate_snax_gen,snax_data_reshuffler))

    VSIM_BENDER += -t snax_data_reshuffler
    VLT_BENDER  += -t snax_data_reshuffler
    VCS_BENDER  += -t snax_data_reshuffler

endif
```

**Review by rgantonio — COMMENTED** [review]
Don't push for now. Leave this for Guilherme.

File: target/snitch_cluster/Makefile (hunk 1 above)
Comment (jorendumoulin) [reply target/snitch_cluster/Makefile:168]: Maybe #112 can avoid all of this code 😸

Comment (xiaoling-yi) [reply target/snitch_cluster/Makefile:168]: Hi @jorendumoulin , we need two targets here, not merely the filename. Do you have any thoughts about this?

**Pull request closed without merging**